### ISSUE TITLE
AttributeFactory generates incorrect type check

### ISSUE THREAD

**Issue by SleepWellPupper** [PaulBraetz/RhoMicro.CodeAnalysis#43]
The type check is inverted, i.e. it fails when the type is met, and succeeds on any other type.
The Metadata name is incorrect, it should be the type name only.
The namespace is obtained incorrectly. It should utilize `ToDisplayString()`.

```cs
if(data.AttributeClass != null 
    && data.AttributeClass.MetadataName == "RhoMicro.CodeAnalysis.JsonSchemaAttribute" 
    && data.AttributeClass.ContainingNamespace.Name == "RhoMicro.CodeAnalysis")
{
    return false;
}
```
should instead be:
```cs
if(data.AttributeClass == null 
    || data.AttributeClass.MetadataName != "JsonSchemaAttribute" 
    || data.AttributeClass.ContainingNamespace.ToDisplayString() != "RhoMicro.CodeAnalysis")
{
    return false;
}
```

**Comment by PaulBraetz** [comment]
Another note:
The Metadata name is incorrect, it should be the type name only:
```cs
if(data.AttributeClass != null 
    && data.AttributeClass.MetadataName == "RhoMicro.CodeAnalysis.JsonSchemaAttribute" 
    && data.AttributeClass.ContainingNamespace.Name == "RhoMicro.CodeAnalysis")
{
    return false;
}
```
should instead be:
```cs
if(data.AttributeClass == null 
    || data.AttributeClass.MetadataName != "JsonSchemaAttribute" 
    || data.AttributeClass.ContainingNamespace.Name != "RhoMicro.CodeAnalysis")
{
    return false;
}
```

**Comment by SleepWellPupper** [comment]
fixed in 16.0.3
